pkg/frontend/querymiddleware/split_by_instant_interval.go

```diff
@@ -150,12 +151,14 @@ func (s *splitInstantQueryByIntervalMiddleware) Do(ctx context.Context, req Requ
 
 	qry, err := newQuery(req, s.engine, lazyquery.NewLazyQueryable(shardedQueryable))
 	if err != nil {
+		level.Warn(log).Log("msg", "failed to create new query from splittable request", "req", req.GetQuery(), "err", err)
 		return nil, apierror.New(apierror.TypeBadData, err.Error())
 	}
 
 	res := qry.Exec(ctx)
 	extracted, err := promqlResultToSamples(res)
 	if err != nil {
+		level.Warn(log).Log("msg", "failed to extract promql results from splittable request", "res", res, "err", err)
```

Comment:
It's more likely the execution failed. I would change the message saying "failed to execute split instant query". Also do not log `res` (will be huge, if serializsed).